vaapi/vaapicontext.cpp

```diff
@@ -90,14 +90,16 @@ YamiStatus VaapiConfig::create(const DisplayPtr& display,
     VAConfigID config;
 
     if (!checkProfileCompatible(display, profile)){
-        ERROR("Unsupport profile");
+        ERROR("Unsupported profile.");
         return YAMI_UNSUPPORTED;
     }
 
     vaStatus = vaCreateConfig(display->getID(), profile, entry, attribList, numAttribs, &config);
 
-    if (!checkVaapiStatus(vaStatus, "vaCreateConfig "))
+    if (!checkVaapiStatus(vaStatus, "vaCreateConfig ")) {
+	ERROR("Unable to create config. This may be due to an unsupported entrypoint.");
```

Comment:
What about saying: "Unable to create config for profile %s and entrypoint %s".  

At this point of execution you know which entrypoint and profile were requested. Then user can run vainfo to confirm the codec is supported.

Reply:
maybe we can return YAMI_UNSUPPORTED here too

Reply:
chivakker, I have some questions about how to go about what you requested.

I don't see a good way to converting the profile / entrypoint to a string with libva or libyami. However, libva-utils/vainfo/vainfo.c does contain such code.

I could copy-and-paste the code from vainfo.c into this file or yamiinfo.cpp, but that seems a little distasteful from a maintenance perspective. Alternatively I could move the code to libva/va/va.c and use it from both projects. There's a function with a similar purpose already in va.c: vaErrorStr( ). However, doing so would add another API to libva which we may not want.

What do you think?

Reply:
@stripes416 IIRC, there should be a ProfileToString utility function and EntrypointToString.  I think it might be inside the test folder.  You could probably copy them instead.

Reply:
I'd prefer to see C++ style string conversions used... see "toString()" and stream operators from intel-vaapi-driver/test/i965_streamable.h.  It's ok to copy that code into this project (e.g. add "vaapi/vaapistreamable.h").

Reply:
@xuguangxin returning YAMI_UNSUPPORTED only makes sense when vaStatus is VA_STATUS_ERROR_UNSUPPORTED_PROFILE or VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT, right?  Otherwise, we should return YAMI_ERROR?

Reply:
@uartie , yeah, you are fright